pennylane/resource/resource.py

```diff
@@ -88,7 +126,14 @@ def _count_resources(tape, shots: int) -> Resources:
     num_gates = 0
     gate_types = defaultdict(int)
     for op in tape.operations:
-        gate_types[op.name] += 1
-        num_gates += 1
+        if isinstance(op, ResourcesOperation):
+             op_resource = op.resources()
+             for d in op_resource.gate_types:
+                 gate_types[d] += op_resource.gate_types[d]
+             num_gates += sum(op_resource.gate_types.values())
+
+         else:
```

Comment:
```suggestion
        else:
```
Slightly out of alignment leading to an error.